packages/shorebird_cli/lib/src/artifact_builder.dart

```diff
@@ -274,6 +274,45 @@ class ArtifactBuilder {
     });
   }
 
+  Future<void> buildLinuxApp({
+    String? flavor,
+    String? target,
+    List<String> args = const [],
+    String? base64PublicKey,
+    DetailProgress? buildProgress,
+  }) async {
+    await _runShorebirdBuildCommand(() async {
+      const executable = 'flutter';
+      final arguments = [
+        'build',
+        'linux',
+        '--release',
+        if (flavor != null) '--flavor=$flavor',
+        if (target != null) '--target=$target',
+        ...args,
+      ];
+
+      final buildProcess = await process.start(
+        executable,
+        arguments,
+        runInShell: true,
```

Comment:
I think you'll want to avoid this whenever you can. :)

Reply:
What this does is pass the arguments array to `sh` rather than to the executable directly, which means they get processed (e.g. variable expansion, quote simplification, etc.) which is sometimes what you want?  But rarely.